### PR TITLE
EUI-2129 Grouping Form Controls

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2129

### Change description ###
Grouping Form Controls

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/register/constants/organisationAddress.ts [hunk 1]
```diff
@@ -27,69 +27,79 @@ export const organisationAddress = {
...
-      {
-        input: {
-          label: {
-            text: 'Town or city',
-            classes: 'govuk-label--m',
+          {
+            input: {
+              validationError: {
+                value: 'Enter the length of hearing in minutes, for example "20"',
+                controlId: 'officeAddressTwo',
```
Comment (paul-graham) [inline src/register/constants/organisationAddress.ts:56]: The `value` and `controlId` don't seem to match...?

File: src/register/constants/organisationAddress.ts [hunk 2]
```diff
@@ -27,69 +27,79 @@ export const organisationAddress = {
...
+          {
+            input: {
+              label: {
+                text: 'Postcode',
+                classes: 'govuk-label--m',
+              },
+              control: 'postcode',
+              validators: ['required'],
+              validationError: {
+                value: 'Enter enter postcode',
```
Comment (paul-graham) [inline src/register/constants/organisationAddress.ts:96]: One 'enter' too many.

File: src/register/constants/organisationAddress.ts [hunk 3]
```diff
@@ -27,69 +27,79 @@ export const organisationAddress = {
...
+                controlId: 'Poscode',
```
Comment (paul-graham) [inline src/register/constants/organisationAddress.ts:97]: Should match the value of `input.control`, which is `'postcode'`.

File: src/register/constants/pbaNumber.ts [hunk 4]
```diff
@@ -19,43 +19,53 @@ export const pbaNumber = {
...
-          },
-          control: 'PBAnumber1',
-          validators: ['pbaNumberPattern', 'pbaNumberMaxLength', 'pbaNumberMinLength'],
-          validationError: {
-            value: 'Enter a valid PBA number',
-            controlId: 'PBAnumber1',
+        fieldset: [
+          {
+            legend: {
+              text: 'What\'s the Payment by Account (PBA) number for your organisation?',
```
Comment (paul-graham) [inline src/register/constants/pbaNumber.ts:25]: Given the fields below, this phrasing seems off - "What's" is short for "What is", which is a question about a single item, yet there are two PBA fields below. This should be re-phrased.

File: src/register/constants/pbaNumber.ts [hunk 5]
```diff
@@ -19,43 +19,53 @@ export const pbaNumber = {
...
-          label: {
-            text: 'PBA number 2 (optional)',
-            classes: 'govuk-label--m',
+          {
+            input: {
+              label: {
+                text: 'PBA number 1 (optional)',
+                classes: 'govuk-label--m',
+              },
+              control: 'PBAnumber1',
```
Comment (paul-graham) [inline src/register/constants/pbaNumber.ts:35]: Naming convention is a bit off here. Is that what the API dictates? If not, `'pbaNumber1'` or, at a push, `'PBANumber1'` would be more appropriate.

File: src/register/constants/pbaNumber.ts [hunk 6]
```diff
@@ -19,43 +19,53 @@ export const pbaNumber = {
...
-          validationError: {
-            value: 'Enter a valid PBA number',
-            controlId: 'PBAnumber2',
+          {
+            input: {
+              label: {
+                text: 'PBA number 2 (optional)',
+                classes: 'govuk-label--m',
+              },
+              control: 'PBAnumber2',
```
Comment (paul-graham) [inline src/register/constants/pbaNumber.ts:50]: Same thing about the naming convention: `'pbaNumber2'` (or `'PBANumber2'` if that's the pattern you've gone for, above.)

**Review by paul-graham — CHANGES_REQUESTED** [review]

File: src/register/constants/organisationAddress.ts (hunk 1 above)
Comment (RiteshHMCTS) [reply src/register/constants/organisationAddress.ts:56]: rectified

File: src/register/constants/organisationAddress.ts (hunk 2 above)
Comment (RiteshHMCTS) [reply src/register/constants/organisationAddress.ts]: rectified

File: src/register/constants/organisationAddress.ts (hunk 3 above)
Comment (RiteshHMCTS) [reply src/register/constants/organisationAddress.ts]: changed to postcode

File: src/register/constants/pbaNumber.ts (hunk 4 above)
Comment (RiteshHMCTS) [reply src/register/constants/pbaNumber.ts]: rectified

File: src/register/constants/pbaNumber.ts (hunk 5 above)
Comment (RiteshHMCTS) [reply src/register/constants/pbaNumber.ts]: changed to PBANumber1

File: src/register/constants/pbaNumber.ts (hunk 6 above)
Comment (RiteshHMCTS) [reply src/register/constants/pbaNumber.ts]: changed to PBANumber2

File: src/register/constants/organisationAddress.ts [hunk 7]
```diff
@@ -27,69 +27,79 @@ export const organisationAddress = {
...
+            input: {
+              label: {
+                text: 'Postcode',
+                classes: 'govuk-label--m',
+              },
+              control: 'postcode',
+              validators: ['required'],
+              validationError: {
+                value: 'Enter postcode',
+                controlId: 'Postcode',
```
Comment (paul-graham) [inline src/register/constants/organisationAddress.ts:97]: Needs to be `'postcode'` with a lowercase 'p' as that's what the `control` is on line 93.

**Review by paul-graham — CHANGES_REQUESTED** [review]
The `'postcode'` one is still not _quite_ right. :)

File: src/register/constants/organisationAddress.ts (hunk 7 above)
Comment (RiteshHMCTS) [reply src/register/constants/organisationAddress.ts]: sorry; changed it to postcode

**Review by paul-graham — DISMISSED** [review]
Thanks for making the requested changes, @RiteshHMCTS. Approved. 👍

**Review by adnanakgun — DISMISSED** [review]

**Review by paul-graham — APPROVED** [review]
All still looks fine. Re-approved. 👍

**Review by adnanakgun — APPROVED** [review]

**Review by paul-graham — DISMISSED** [review]

**Review by paul-graham — APPROVED** [review]

**Review by adnanakgun — APPROVED** [review]

**Pull request closed without merging**